src/GraphQL/Types/Schema.cs

```diff
@@ -451,7 +451,7 @@ IEnumerable<ISchemaNodeVisitor> GetVisitors()
             {
                 // At this point, Initialized will return false, and Initialize will still lock while waiting for initialization to complete.
                 // However, AllTypes and similar properties will return a reference to SchemaTypes without waiting for a lock.
-                _allTypes.ApplyMiddleware(FieldMiddleware);
+                _allTypes.ApplyMiddleware(FieldMiddleware, this);
```

Comment:
I don't mind the added method, but I dislike that we are calling an internal method.  Better to mark the old method as obsolete and expose the new method publicly.

Reply:
I reverted it.